edge/cmd/edgecore/app/server.go

```diff
@@ -138,12 +138,10 @@ func environmentCheck() error {
 	}
 
 	for _, process := range processes {
-		// if kubelet is running, return error
-		if process.Executable() == "kubelet" {
+		switch process.Executable() {
+		case "kubelet": // if kubelet is running, return error
 			return errors.New("kubelet should not running on edge node when running edgecore")
-		}
-		// if kube-proxy is running, return error
-		if process.Executable() == "kube-proxy" {
+		case "kube-proxy": // if kube-proxy is running, return error
```

Comment:
There are only these two situations, I think there is no need to use `switch` to extend the judgment.

Reply:
Yes, but for each process in processes, the former calls process.Executable() twice while the right only calls once.

Reply:
```golang
func (p *UnixProcess) Executable() string {
	return p.binary
}
```
This method directly returns the structure field without additional processing logic.

Reply:
If you think the code is so beautiful, I will quit. We should never talk about the detail of `Executable()` here, it should be transparent to the code what we talk about.

Reply:
Switch and if all fine to me :)